question-servers/freeformPythonLib/prairielearn.py

```diff
@@ -913,85 +925,13 @@ def string_to_2darray(s, allow_complex=True):
                     A[i, j] = ans
                 except Exception:
                     # Return error if entry could not be converted to float or complex
-                    return (None, {'format_error': 'Entry ({:d}, {:d}) of matrix "{:s}" has invalid format.'.format(i + 1, j + 1, s_row[i][j])})
+                    return (None, {'format_error': f'Entry "{escape_unicode_string(s_row[i][j])}" at ({i + 1}, {j + 1}) of matrix has invalid format.'})
```

Comment:
I guess we should treat `format_errors` as HTML? So we could do something like this:

```suggestion
                    return (None, {'format_error': f'Entry <code class="user-output-invalid">{html.escape(escape_unicode_string(s_row[i][j]))}</code> at ({i + 1}, {j + 1}) of matrix has invalid format.'})
```

Then in the popover template in `pl-matrix-input.mustache` we would use triple-braces on `{{{format_error_message}}}`?

I guess this will require checking whether we need to put `html.escape()` around any other format error messages?

Reply:
Ah so actually it is trivial after all -- I wasn't aware of `{{{`. But I also gave it up quickly because it felt a bit code-smelly to put the markup in server code 😵

Reply:
Yep, I agree with you on the code smell. But now that I think about it further I think we probably do want to be able able to put HTML in `format_errors`, because they are fundamentally for display to the student and being able to style the errors can be helpful in making them clearer. Unfortunately it means we need to be careful with escaping! 😝